indico/modules/events/management/client/js/EmailAuthors.jsx

```diff
@@ -0,0 +1,211 @@
+  const [selectedUsers, setSelectedUsers] = useState([]);
+  const [successMessage, setSuccessMessage] = useState(undefined);
+  const [activeFilters, setActiveFilters] = useState([]);
+  const formData = new FormData();
+  sourceIds.forEach(v => formData.append(`${context}_id`, v));
+  const {data, loading} = useIndicoAxios(
+    {
+      url: CONTEXTS[context].personListURL({event_id: eventId}),
+      method: 'POST',
+      data: formData,
+      headers: {'content-type': 'multipart/form-data'},
+    },
+    {camelize: true}
+  );
+  const extraRoles =
+    context === 'abstract'
+      ? [
+          {
+            icon: 'user-check icon',
+            isActive: p => p.submitter,
+            titleActive: Translate.string('This person submitted an abstract'),
+            titleInactive: Translate.string("This person hasn't submitted any abstracts"),
+          },
+        ]
+      : undefined;
+
+  const onChangeSelection = ({person, user}) => {
+    person && setSelectedPersons(person);
+    user && setSelectedUsers(user);
+  };
+
+  const isVisible = person =>
+    !activeFilters.length || activeFilters.every(filter => FILTER_OPTIONS[filter].isMatch(person));
+
+  const toggleFilter = name => {
+    setActiveFilters(
+      activeFilters.includes(name)
+        ? activeFilters.filter(f => f !== name)
+        : [...activeFilters, name]
+    );
+  };
+
+  const filterButtons = Object.entries(FILTER_OPTIONS)
+    .filter(([, filter]) => !filter.contexts || filter.contexts.includes(context))
+    .map(([name, filter]) => ({
+      key: name,
+      content: filter.text,
+      onClick: () => toggleFilter(name),
+      active: activeFilters.includes(name),
+    }));
+
+  return (
+    <Modal open onClose={() => onClose()} size="large">
+      <Modal.Header content={Translate.string('Email authors')} />
+      <Modal.Content>
+        {successMessage}
+        <div>
+          <Translate
+            as="div"
+            style={{display: 'inline-block', fontWeight: 'bold', padding: '0.5em'}}
+          >
+            Filter by
+          </Translate>
+          <Button.Group buttons={filterButtons} compact />
+        </div>
+        {loading || !data ? (
+          <Dimmer active>
+            <Loader />
+          </Dimmer>
+        ) : data.eventPersons.length ? (
+          <PersonList
+            persons={data.eventPersons}
+            selectedPersons={selectedPersons}
+            selectedUsers={selectedUsers}
+            onChangeSelection={onChangeSelection}
+            isSelectable={person => !!person.email}
+            isVisible={isVisible}
+            extraRoles={extraRoles}
+          />
+        ) : (
+          <Message error>{CONTEXTS[context].emptyListMessage}</Message>
+        )}
+      </Modal.Content>
+      <Modal.Actions>
+        <EmailButton
+          eventId={eventId}
+          personIds={selectedPersons.filter(id =>
+            isVisible(data.eventPersons.filter(p => p.id === id)[0])
```

Comment:
`data.eventPersons.find(p => p.id === id)`

but maybe using a Map would be better? if you select everything I think you get O(n²) here...